crates/duckdb/src/lib.rs

```diff
@@ -357,132 +257,166 @@ pub fn version() -> &'static str {
     env!("CARGO_PKG_VERSION")
 }
 
+fn to_geoarrow_record_batch(mut record_batch: RecordBatch) -> Result<RecordBatch> {
+    if let Some((index, _)) = record_batch.schema().column_with_name("geometry") {
+        let geometry_column = record_batch.remove_column(index);
+        let binary_array: GenericByteArray<GenericBinaryType<i32>> =
```

Comment:
Ideally you should check here whether the geometry column uses `i32` or `i64` offsets. _Probably_ duckdb only ever exports `i32` arrays, so you're probably ok.

Reply:
Worth checking though to be sure, thanks.